### ISSUE TITLE
Lag is reported for deleted consumer group, if it uses only assign() API

### ISSUE THREAD

**Issue by emitskevich-blp** [linkedin/Burrow#818]
**Summary:**
If consumer group uses `assign()` API and never calls `subscribe()`, Burrow will continue report lag alarms even after explicit Delete Group request to Kafka.

**Technicals:**
Kafka, when processing Delete Group request, makes 2 things: 
1. Puts partition-level tombstone on every partition related to consumer group to drop stored offsets: [link](https://github.com/emitskevich-blp/kafka/blob/da3304ecb6f2e41de5891ee1128a8aba7330f8bd/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L890-L894)
2. Additionally puts group-level tombstone only if `generation > 0`: [link](https://github.com/emitskevich-blp/kafka/blob/da3304ecb6f2e41de5891ee1128a8aba7330f8bd/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L897-L906)

Burrow, reading these tombstones, [stops monitoring consumer group when sees group-level tombstone](https://github.com/linkedin/Burrow/blob/8690c5844cab06601634bf21fa07591f951068fc/core/internal/consumer/kafka_client.go#L516-L528), but [ignores partition-level tombstones](https://github.com/linkedin/Burrow/blob/8690c5844cab06601634bf21fa07591f951068fc/core/internal/consumer/kafka_client.go#L449-L453).

When consumers only use `assign()` API, which is valid approach, generation is always zero. It makes Burrow miss "delete group" request.

**Expected behavior:**
Burrow stops monitoring consumer group after meeting tombstone for all associated partitions, even if no group-level tombstone is present.

**Comment by emitskevich-blp** [comment]
I'm looking to make a contribution to fix it, but I'd like to have an initial feedback on the approach. 

The plan is to have a counter for the num of associated partitions, decrease it on partition-level tombstone, then call `StorageSetDeleteGroup` when counter is zero.

**Comment by emitskevich-blp** [comment]
@bai could you check the issue itself and validate the suggested approach?

**Comment by bai** [comment]
Hey @emitskevich-blp - thanks for looking into this. This is a sound approach, would appreciate if you could submit a PR.

**Comment by emitskevich-blp** [comment]
Looks like groups reaper (#657) is intended to cover the case. I researched described behavior in the env with groups reaper disabled (default). My observations now show it works fine with reaper enabled, since Kafka itself knows when such groups are deleted.

@d1egoaz (#657 author) could you confirm the statements in this comment by any chance?

**Comment by d1egoaz** [comment]
> kafka itself knows when such groups are deleted.

Yeah, that's the best way to do it. Let Kafka handle this `ListConsumerGroups`, and we'll take care of removing it from the Burrow side.

Great technical analysis, by the way!
